### PR TITLE
Docker-Compose: Avoid setting hostname to ''

### PR DESCRIPTION
**Pull request by glensc**
The hostname is literally set to `''`:

Before:
```
# docker-compose exec smtp hostname
''
```

After:
```
# docker-compose exec smtp hostname
c531c19841de
```

### CODE REVIEW

**Review by aminvakil — CHANGES_REQUESTED** [review]
I think there was a reason this has been added in https://github.com/getsentry/self-hosted/pull/1116/commits/6c92774fae3a76df266cc035a7f5c7466c441b6f which was in https://github.com/getsentry/self-hosted/pull/1116.

Investigating further...

**Comment by aminvakil** [comment]
I just tested your change with docker-compose 2.3.2 and after this, `snuba` container keeps restarting without any log.

**Comment by glensc** [comment]
@aminvakil in any case, the used syntax is incorrect. it's already well known (to me at least) that docker compose doesn't like extra quoting. it's not shell script. for example if you place a variable with spaces to .env YOU MAU NOT QUOTE THE VALUE, or you get quotes in env value itself.

anyway, after my fix, if env is missing the valuer is set to empty (rather apostrhophes), and then default hostname will be set, which is generated name. as seen in example I provided

**Comment by glensc** [comment]
I fail to see how "smtp" container "hostname" is related to "snuba" container :D

If you insist on quoting, then this might work (not tested):

```yml
  smtp:
    <<: *restart_policy
    image: tianon/exim4
    hostname: "${SENTRY_MAIL_HOST:-}"
```

but that's just yaml syntax, shouldn't affect the variables expansion.

**Comment by aminvakil** [comment]
> I just tested your change with docker-compose 2.3.2 and after this, `snuba` container keeps restarting without any log.

Ok, reverting this PR change results in `sentry-self-hosted-snuba-subscription-consumer-transactions-1` container restarting over and over again.

**Comment by aminvakil** [comment]
> I fail to see how "smtp" container "hostname" is related to "snuba" container :D
> 
> If you insist on quoting, then this might work (not tested):
> 
> ```yaml
>   smtp:
>     <<: *restart_policy
>     image: tianon/exim4
>     hostname: "${SENTRY_MAIL_HOST:-}"
> ```
> 
> but that's just yaml syntax, shouldn't affect the variables expansion.

Yeah, it was a mistake, it's not related. But I still want to understand why it has been added at the first place.

**Comment by BYK** [comment]
GitHub did something to how they use Docker across matrix jobs and I think this is causing issues. We may want to scope the project name with the job name in docker-compose setups to avoid the issues. The failure on 1.29.2 seems related to that.

**Comment by aminvakil** [comment]
@glensc As https://github.com/getsentry/self-hosted/commit/de81299bafe2e36e9660f31837569f93b5788da1 has been merged, can you please rebase your branch to retrigger the CI?

It should pass now.

**Comment by glensc** [comment]
@aminvakil, you can do that yourself, really:

image

**Comment by aminvakil** [comment]
@glensc Do you see "maintainer" here?

image

**Review by BYK — APPROVED** [review]

**Review by aminvakil — APPROVED** [review]
LGTM.

Closing and reopening to trigger the CI.

File: docker-compose.yml
Comment (chadwhitacre) [inline docker-compose.yml:96]: lgtm

> Additionally when using the [2.1 file format](https://docs.docker.com/compose/compose-file/compose-versioning/#version-21), it is possible to provide inline default values using typical shell syntax:
>
> - `${VARIABLE:-default}` evaluates to `default` if `VARIABLE` is unset or empty in the environment.

https://docs.docker.com/compose/environment-variables/

I'll take your word for it on quoting, @glensc. :P

**Review by chadwhitacre — APPROVED** [review]

**Comment by chadwhitacre** [comment]
Sadly needs another rebase once #1375 lands.

**Comment by chadwhitacre** [comment]
I landed #1375. Can you rebase again @glensc? 🙏

**Comment by glensc** [comment]
image

Why do you need me to do the rebases? Makes only useless ping-pongs!

File: docker-compose.yml
Comment (glensc) [reply docker-compose.yml:96]: No need to take on anything, just see the proof in PR body.

**Comment by glensc** [comment]
Build flackyness due to multiple versions in same docker instance still not resolved:

- https://github.com/getsentry/self-hosted/runs/5540753414?check_suite_focus=true

```
 Container self-hosted-0-nginx-1  Recreated
  Error response from daemon: Renaming a container with the same name as its current name
  ++ teardown ERR 45
```

and unrelated to this PR.

**Comment by chadwhitacre** [comment]
> Build flakiness due to multiple versions in same docker instance still not resolved:

https://github.com/getsentry/self-hosted/issues/1294 ftr

> The fix didn't apparently work

de81299bafe2e36e9660f31837569f93b5788da1 was for #1341, not #1294. We have [several open CI issues](https://github.com/getsentry/self-hosted/issues?q=is%3Aissue+is%3Aopen+ci+flakes). 😞

**Comment by glensc** [comment]
I thought this was to solve container name collisions:

> `COMPOSE_PROJECT_NAME: self-hosted-${{ strategy.job-index }}`

**Comment by glensc** [comment]
hey. CI green. merge fast before someone requests rebase! :D

**Comment by chadwhitacre** [comment]
Close one! Thanks @glensc! 😁

**Pull request merged**
